src/cycles/eulerian.jl

```diff
@@ -0,0 +1,74 @@
+# Checking for existence of Eulerian circuit and Eulerian trail
+
+function multinode_component_count end
+@traitfn function multinode_component_count(
+    graph::AG::(!IsDirected),
+    ) where {T, AG <: AbstractGraph{T}}
+
+    components = connected_components(graph)
+
+    count = 0
+    for component in components
```

Comment:
I just realised that you can ignore isolated vertices. So your original solution might have been better, because you don't have to allocate memory for the connected_components. Sorry about that. 
But you have to be a bit careful. It could be, that a component consists of a single vertex, but has a self-loop attached to this vertex. In that case, I think, you also have to increase `count`.

Reply:
Okay, I actually referred a different resource which seems do not considered for self-loop, now I will strictly follow https://en.wikipedia.org/wiki/Eulerian_path because I think conditions here are appropriate and cover all cases.

Reply:
But in the earlier solution I still user a visited array so they still account in memory, any suggestion for this?